hashes/zkevm-keccak/src/keccak_packed_multi.rs

```diff
@@ -287,6 +285,31 @@ impl<F: FieldExt> CellManager<F> {
         } else {
             assert!(column_idx == self.columns.len());
             let advice = meta.advice_column();
+            // This is the column where the input 64-bit words are stored.
+            if let Ok(num_flex_advice) = var("FLEX_GATE_NUM_ADVICE").map(|s| {
+                s.parse::<usize>().expect("Cannot parse FLEX_GATE_NUM_ADVICE env var as usize")
+            }) {
+                let col_to_enable =
+                    if num_flex_advice == 1 { num_flex_advice + 1 } else { num_flex_advice + 2 };
+                if advice.index() == col_to_enable {
+                    meta.enable_equality(advice);
+                }
+            }
+            // This is the column where the output bytes are stored. It happens at the last keccak column,
+            // whose index will be shifted by the number of flex gate columns.
+            if let Ok(last_keccak_column) = var("LAST_KECCAK_COLUMN").map(|s| {
+                s.parse::<usize>().expect("Cannot parse LAST_KECCAK_COLUMN env var as usize")
+            }) {
+                if let Ok(num_flex_advice) = var("FLEX_GATE_NUM_ADVICE").map(|s| {
+                    s.parse::<usize>().expect("Cannot parse FLEX_GATE_NUM_ADVICE env var as usize")
+                }) {
+                    let col_to_enable = last_keccak_column
+                        + if num_flex_advice == 1 { num_flex_advice } else { num_flex_advice + 1 };
+                    if advice.index() == col_to_enable {
+                        meta.enable_equality(advice);
+                    }
+                }
+            }
```

Comment:
These two blocks are pretty similar, there's probably a function that could be abstracted out and called here. Maybe in the docs of that function you could elaborate a bit on what's going on, because I don't quite follow.